src/builder/QuarkBuilder.sol

```diff
@@ -74,22 +73,28 @@ contract QuarkBuilder {
 
         // XXX confirm that the user is not withdrawing beyond their limits
 
-        assertFundsAvailable(
-            repayIntent.chainId, repayIntent.assetSymbol, repayIntent.amount, chainAccountsList, payment
-        );
+        bool isMaxRepay = repayIntent.amount == type(uint256).max;
+        bool useQuotecall = isMaxRepay;
```

Comment:
this should always be true, right? whenever we're doing a max repay we should use Quotecall

Reply:
I think it's more like...
```
Case 1: 
User owed 100 USDC to Comet. User hold 50USDC. 
User use maxRepay, it used out all user's 50USDC to pay the debt and make it owe 50USDC at the end. 

In this case Quotecall is better. (Similar to transferMax to Comet.)

Case 2: 
User owe 100USDC to comet. User hold 150USDC.
User use RepayMax, it uses user holding of 100USDC and payoff all the debt at Comet and leaving 50USDC in wallet.

In this case use Paycall is better, as there is no need to drain some dust from the wallet to make user feel "empty" wallet.  (User still has 50USDC in the wallet anyway.)
```

To decide when to use quotecall, I think mostly goes down to whether the use case we want user to feel better when they see their "emptied" wallet.

Reply:
Wouldn't Case 1 revert? If you called `maxRepay` for a 100 USDC borrow position but you only had 50 USDC?

Reply:
If only case 2, then probably just paycall should be good. Cuz Case 2 we don't have to make user feel "emptied" 🤔 

Was thinking about builder to smart switch between apply all user owed position v.s. apply user maxBalance if user try to do maxRepay when not having all the fund.

Reply:
Weirdly enough, I think this is a case where we always use `Paycall`. In transfer max, all the funds leave the user's wallet, so `Quotecall` is used to not leave any dust. In repay max, the user is not transferring all the funds in their wallet. They are transferring just enough to pay back their entire borrow. So `Quotecall` here doesn't actually achieve anything.